gt/main.c

```diff
@@ -126,6 +130,10 @@ gt_parse_incoming_pkt(struct rte_mbuf *pkt, struct gt_packet_headers *info)
 		info->inner_ip_ver = ETHER_TYPE_IPv4;
 		info->l4_proto = inner_ipv4_hdr->next_proto_id;
 		info->l4_hdr = &inner_ipv4_hdr[1];
+		/* Copy outer header's ECN bits to inner header. */
+		inner_ipv4_hdr->type_of_service =
+			(inner_ipv4_hdr->type_of_service & ~IPTOS_ECN_MASK) |
+			ecn;
```

Comment:
This function is called `gt_parse_incoming_pkt()`, that is, it's a surprise that besides parsing the incoming packet it's also changing the header of the packet. Isn't there a way to avoid this surprise?